k8s/upgrade-job/src/upgrade/data_plane.rs

```diff
@@ -158,124 +221,137 @@ async fn wait_for_rebuild(rest_client: &RestClientSet) -> Result<()> {
 }
 
 /// Issue the node drain command on the node.
-async fn issue_node_drain(node_name: &str, rest_client: &RestClientSet) -> Result<()> {
-    rest_client
-        .nodes_api()
-        .put_node_drain(node_name, DRAIN_FOR_UPGRADE)
-        .await
-        .context(DrainStorageNode {
-            node_name: node_name.to_string(),
-        })?;
+async fn drain_storage_node(node_id: &str, rest_client: &RestClientSet) -> Result<()> {
+    let drain_label_for_upgrade: String = DRAIN_FOR_UPGRADE.to_string();
 
-    tracing::info!(node.name = %node_name, "Drain started for {PRODUCT} Node");
+    loop {
+        let storage_node =
+            rest_client
+                .nodes_api()
+                .get_node(node_id)
+                .await
+                .context(GetStorageNode {
+                    node_id: node_id.to_string(),
+                })?;
 
-    Ok(())
+        match storage_node
+            .clone()
```

Comment:
clone not needed?

Reply:
Yes, it does. NodeSpec Is moved into each loop iteration.

Reply:
As it turns out, it's not need... :)